### PR TITLE
Format code with mix format

### PR DESCRIPTION
**Pull request by axelson**
And add a .formatter.exs file

This is basically a redo of #13 but now there should be less conflicts since there is only 1 outstanding pull request (which only touches 4 files)

### CODE REVIEW

**Comment by msaraiva** [comment]
@axelson all PRs merged! Feel free to update this and merge whenever you want.

File: lib/elixir_sense/core/ast.ex [hunk 1]
```diff
@@ -59,10 +79,12 @@ defmodule ElixirSense.Core.Ast do
...
 
   defp add_directive_modules_to_env(env, directive, modules) do
-    directive_string = modules
-    |> Enum.filter(&Code.ensure_compiled?/1)
-    |> Enum.map(&"#{directive} #{Introspection.module_to_string(&1)}")
-    |> Enum.join("; ")
+    directive_string =
+      modules
+      |> Enum.filter(&Code.ensure_compiled?/1)
+      |> Enum.map(&"#{directive} #{Introspection.module_to_string(&1)}")
```
Comment (asummers) [inline lib/elixir_sense/core/ast.ex]: For future PR: This should be a map_join/3 for perf reasons.

File: lib/elixir_sense/core/state.ex [hunk 2]
```diff
@@ -152,60 +151,71 @@ defmodule ElixirSense.Core.State do
...
-    |> Enum.join(@or_marker)
-    |> String.to_atom()
+    joined_implementations =
+      implementations
+      |> Enum.map(fn parts ->
+        parts
+        |> Enum.map(&Atom.to_string/1)
+        |> Enum.join(@dot_marker)
+      end)
+      |> Enum.join(@or_marker)
```
Comment (asummers) [inline lib/elixir_sense/core/state.ex]: Also fot future PR, these should both be map_joins for performance reasons.

**Comment by msaraiva** [comment]
@asummers yes, I think it's a good idea 👍

**Comment by axelson** [comment]
I ended up doing a force-push because dealing with the merge conflicts wasn't quite working. Also added the `mix format --check-formatted` to travis.

**Comment by msaraiva** [comment]
@axelson awesome! 🎉

**Pull request merged**